UM/Qt/Bindings/ProfilesModel.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2015 Ultimaker B.V.
+# Uranium is released under the terms of the AGPLv3 or higher.
+
+from UM.Qt.ListModel import ListModel
+from UM.Application import Application
+from UM.Message import Message
+from UM.Settings.Profile import Profile
+
+from PyQt5.QtCore import Qt, pyqtSlot, pyqtSignal, pyqtProperty, QUrl
+
+from UM.i18n import i18nCatalog
+catalog = i18nCatalog("uranium")
+
+class ProfilesModel(ListModel):
+    IdRole = Qt.UserRole + 1
+    NameRole = Qt.UserRole + 2
+    ActiveRole = Qt.UserRole + 3
+    ReadOnlyRole = Qt.UserRole + 4
+
+    def __init__(self, parent = None):
+        super().__init__(parent)
+
+        self._select_global = False
+
+        self.addRoleName(self.IdRole, "id")
+        self.addRoleName(self.NameRole, "name")
+        self.addRoleName(self.ActiveRole, "active")
+        self.addRoleName(self.ReadOnlyRole, "readOnly")
+
+        self._manager = Application.getInstance().getMachineManager()
+
+        self._manager.profilesChanged.connect(self._onProfilesChanged)
+        self._manager.activeProfileChanged.connect(self._onActiveProfileChanged)
+        self._manager.profileNameChanged.connect(self._onProfileNameChanged)
+        self._onProfilesChanged()
+
+    selectGlobalChanged = pyqtSignal()
+
+    def setSelectGlobal(self, select):
```

Comment:
I'd change this name. It's confusing what it means. (eg; Is global selected? Is the selection global?)